### PR TITLE
Fix validator activation monitoring with inactive keys

### LINKED ISSUE #8546: Importing validators still requires restart to pick up newly-imported
[logs.txt](https://github.com/prysmaticlabs/prysm/files/6073281/logs.txt)
[metrics-and-wallet.txt](https://github.com/prysmaticlabs/prysm/files/6073282/metrics-and-wallet.txt)

# 🐞 Bug Report

### Description

Validators imported while no validators are active still aren't being processed.

### Has this worked before in a previous version?

No.  But behavior is getting better.

## 🔬 Minimal Reproduction

Import new validators while waiting for activation.

## 🔥 Error

<pre><code>

</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>

  </code>
</pre>

**What version of Prysm are you running? (Which release)**

v1.2.1
<pre>
  <code>

  </code>
</pre>

**Anything else relevant (validator index / public key)?**

I've attached logs / output from a client reporting 14 validators when the wallet has 79.

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

When starting a Prysm validator, the client would monitor the beacon chain for activations of its public keys. If none of the keys are active, it will continue to monitor until one of the keys has been activated. One of the features of the validator is that the key manager is monitoring for new key additions. The bug was that the new key additions were not being included in the activation monitoring routine and would require a restart to pick up the changes.

Tested in pyrmont and saw the keys reloaded.
```
[2021-03-04 20:18:58]  INFO validator: Waiting for deposit to be observed by beacon node pubKey=0xb2511b2557bc status=UNKNOWN_STATUS
[2021-03-04 20:19:10]  INFO validator: Waiting for deposit to be observed by beacon node pubKey=0xb2511b2557bc status=UNKNOWN_STATUS
[2021-03-04 20:19:16]  INFO imported-keymanager: Reloaded validator keys into keymanager
[2021-03-04 20:19:22]  INFO validator: Waiting for deposit to be observed by beacon node pubKey=0xb2511b2557bc status=UNKNOWN_STATUS
[2021-03-04 20:19:22]  INFO validator: Waiting for deposit to be observed by beacon node pubKey=0x87f88acc6982 status=UNKNOWN_STATUS
```

**Which issues(s) does this PR fix?**

Fixes #8546 

**Other notes for review**

Relevant doc: https://docs.google.com/document/d/1fLdUi6AHQ5-wYOy6DCzW5JlLlq3VuI275-KoyEZ2Pmw/edit#
This PR solves the following issues:
- [x] Inactive Add Active
- [x] Inactive Add Inactive 

Some unrelated files are included due to goimports changes.

This PR also properly stops the SubscribeAccountChanges goroutine.

### CODE REVIEW

File: validator/client/wait_for_activation.go [hunk 1]
```diff
@@ -129,3 +148,24 @@ func incrementRetries(ctx context.Context) context.Context {
 	attempts := streamAttempts(ctx)
 	return context.WithValue(ctx, waitForActivationAttemptsContextKey, attempts+1)
 }
+
+func handleAccountsChanged(ctx context.Context, v Validator, accountsChangedChan chan<- struct{}) {
```
Comment (nisdas) [inline validator/client/wait_for_activation.go:152]: This method seems redundant, its basically creates a channel to relay account changes to the `accountsChangedChan` . How about changing the `accountsChangedChan` type to a ` chan<- [][48]byte` and then subscribe account changes to it directly ? You could skip this additional routine and channel being created here.

File: validator/client/wait_for_activation.go [hunk 2]
```diff
@@ -19,8 +19,27 @@ import (
...
+	go handleAccountsChanged(accountsCtx, v, accountsChangedChan)
+
+	return v.waitForActivation(ctx, accountsChangedChan)
+}
+
+// waitForActivation performs the following:
+// 1) While the key manager is empty, poll the key manager until some validator keys exist.
+// 2) Open a server side stream for activation events against the given keys.
+// 3) In another go routine, the key manager is monitored for updates and emits an update event on
+// the accountsChangedChan. When a event signal is received, restart the waitForActivation routine.
```
Comment (rkapka) [inline validator/client/wait_for_activation.go:38]:
```suggestion
// the accountsChangedChan. When an event signal is received, restart the waitForActivation routine.
```

File: validator/client/wait_for_activation.go [hunk 3]
```diff
@@ -19,8 +19,27 @@ import (
...
+// 4) If the stream is reset in error, restart the routine.
+// 5) If the stream returns a response indicating one or more validators are active, exit this
+// the routine.
```
Comment (rkapka) [inline validator/client/wait_for_activation.go:41]:
```suggestion
// 5) If the stream returns a response indicating one or more validators are active, exit the routine.
```

**Review by rkapka — DISMISSED** [review]

**Review by rkapka — APPROVED** [review]

File: validator/client/wait_for_activation.go (hunk 1 above)
Comment (prestonvanloon) [reply validator/client/wait_for_activation.go:151]: Excellent feedback! Doing this now.

**Review by rkapka — APPROVED** [review]

**Review by rkapka — APPROVED** [review]
looks good

**Comment by prestonvanloon** [comment]
Some feedback from @rkapka in discord is that we could refactor this further to remove the recursive functionality. 
A stream could be restarted in the same method, without adding another recursive call to the stack.

This feature / refactoring will come in a future PR. Thanks!

**Pull request merged**